### PR TITLE
Use optExtensions in Session loader

### PR DESCRIPTION
**Pull request by pepeiborra**
This is a minor correctness fix

### CODE REVIEW

File: session-loader/Development/IDE/Session.hs [hunk 1]
```diff
@@ -380,18 +381,18 @@ data TargetDetails = TargetDetails
...
+             -> [String]            -- ^ extensions to consider
              -> TargetId
              -> IdeResult HscEnvEq
              -> DependencyInfo
              -> IO [TargetDetails]
 -- For a target module we consider all the import paths
-fromTargetId is (TargetModule mod) env dep = do
+fromTargetId is exts (TargetModule mod) env dep = do
     let fps = [i </> moduleNameSlashes mod -<.> ext | ext <- exts, i <- is ]
-        exts = ["hs", "hs-boot", "lhs"]
```
Comment (ndmitchell) [inline session-loader/Development/IDE/Session.hs:390]: Normally we set optExtensions to be `hs` and `lhs`. Before that, we were looking for `hs-boot` too. Is that behaviour deliberate? Was looking for `hs-boot` wrong? If looking for `hs-boot` wasn't wrong, why wasn't `lhs-boot` on the list?

Comment (pepeiborra) [reply session-loader/Development/IDE/Session.hs:390]: Yes, that's a good point. I think "hs-boot" needs to be added to the list

Comment (pepeiborra) [reply session-loader/Development/IDE/Session.hs:390]: I don't know what's the story with `lhs-boot`. https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/separate_compilation.html does not mention them so perhaps they are not a thinh

Comment (ndmitchell) [reply session-loader/Development/IDE/Session.hs:390]: Fair enough. I wish both boot files AND lhs files were not a thing, so the combination of them is a bit of a monster!

**Review by ndmitchell — APPROVED** [review]

**Pull request merged**